### PR TITLE
mixer library split into separate headers and name consistently

### PR DESCRIPTION
**Pull request by dagar**

### CODE REVIEW

**Comment by dagar** [comment]
This PR started off simply splitting the giant mixer.h header, but I continued with additional cleanup and minor improvements.

File: src/lib/mixer/Mixer.hpp [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+	virtual void 			set_thrust_factor(float val) {}
+
+	/**
+	 * @brief Set airmode. Airmode allows the mixer to increase the total thrust in order to unsaturate the motors.
+	 *
+	 * @param[in]  airmode   Select airmode type (0 = disabled, 1 = roll/pitch, 2 = roll/pitch/yaw)
+	 */
+	virtual void			set_airmode(Airmode airmode) {};
+
+	virtual unsigned		get_multirotor_count()  { return 0; }
```
Comment (mcsauder) [inline src/lib/mixer/Mixer.hpp]: Hi @dagar, do you think this should be bumped to the MultiRotor Mixer? (It appears to me that the helicopter mixer is not utilizing it.)  Everything else is looking pretty good in this PR.

Comment (dagar) [reply src/lib/mixer/Mixer.hpp]: No, unfortunately with the current structure any interaction we want with any mixer has to go through a common interface.

**Comment by dagar** [comment]
This saves a little bit of flash and memory on the px4io.

File: src/lib/mixer/MixerGroup.hpp [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+	 * the position of the servo, in the following form:
+	 *
+	 *   S: <angle (deg)> <normalized arm length> <scale> <offset> <lower limit> <upper limit>
+	 *
+	 * @param buf			The mixer configuration buffer.
+	 * @param buflen		The length of the buffer, updated to reflect
+	 *				bytes as they are consumed.
+	 * @return			Zero on successful load, nonzero otherwise.
+	 */
+	int				load_from_buf(Mixer::ControlCallback control_cb, uintptr_t cb_handle, const char *buf, unsigned &buflen);
```
Comment (jlecoeur) [inline src/lib/mixer/MixerGroup.hpp]: You may implement instead
```
static MixerGroup *from_text(Mixer::ControlCallback control_cb, uintptr_t cb_handle, const char *buf, unsigned &buflen);
```
to be fully consistent with the other mixers?

Comment (dagar) [reply src/lib/mixer/MixerGroup.hpp]: Let's think about that for a follow up. MixerGroup is now just the thing that you go through to interact with the full set of loaded mixers. It's no longer a Mixer.

**Pull request merged**